package/spec/frontend/video/prebuffering_spec.js

```diff
@@ -0,0 +1,65 @@
+import '$support/fakeBrowserFeatures';
+import {prebuffering} from 'frontend/VideoPlayer/prebuffering';
+import {createTimeRange} from "$support/timeRanges";
+import sinon from 'sinon';
+
+
+describe('VideoPlayer', function() {
+  describe('#prebuffering', function() {
+    it('returns promise which resolves after prebuffering desired time', async () => {
+      const player = fakePlayer();
+      const callback = sinon.spy();
+
+      prebuffering(player);
+
+      const options = {
+        secondsToBuffer: 10,
+        secondsToWait: 1
+      }
+      await player.prebuffer(options).then(callback);
+
+      expect(player.buffered).toHaveBeenCalledTimes(5)
+      expect(callback).toHaveBeenCalled();
+    });
+
+    it('aborts prebuffering on pause', function() {
+      const player = fakePlayer();
+      const callback = sinon.spy();
+
+      prebuffering(player);
+
+      player.prebuffer().then(callback);
+      player.pause();
+
+      expect(callback).not.toHaveBeenCalled();
```

Comment:
Have you seen this spec fail? I think it passes for the wrong reason. Since `callback` is invoked asynchronously, it will never have been called right after the `prebuffer` call.